alt-tab-macos/ui/Application.swift

```diff
@@ -61,11 +61,13 @@ class Application: NSApplication, NSApplicationDelegate, NSWindowDelegate {
     }
 
     @objc func showCenteredPreferencesPanel() {
-        showCenteredPanel(preferencesPanel!)
+        showCenteredFrontPanel(preferencesPanel!)
     }
 
-    func showCenteredPanel(_ panel: NSPanel) {
-        panel.center()
+    func showCenteredFrontPanel(_ panel: NSPanel, _ screenOverride: NSScreen? = nil) {
+        let preferredScreen = screenOverride ?? Screen.getPreferredScreen() // use optional provided screen or determine it just in time
+        let centerPosition = NSPoint(x: preferredScreen.visibleFrame.midX - panel.frame.width / 2, y: preferredScreen.visibleFrame.midY - panel.frame.height / 2)
+        panel.setFrameOrigin(centerPosition)
```

Comment:
Nice catch. 😎 

Yeah I've noticed that too. Ive changed it on purpose to be vertically centered cause the native app switcher is not using the offset. But we need the offset for other windows e.g. the PreferencesPanel.

I will make this an argument of this method (if one would like to apply the appleVerticalOffset), make it true by default and apply an visibleFrame based ratio. The PreferencesPanel would like to apply it and ThumbnailsPanel would like to be centered without any offset. Agreed?

Reply:
I prefer `ThumbnailsPanel` being higher-than-centered. Like I said it is a thing I encountered in typography. Book titles and even pages are usually higher than centered. However since I don't know, and couldn't Google, an explanation on *how* that is supposed to be more ergonomic, I guess realizing my position is weak. I'll let you decide.

Note that HyperSwitch does centered but WindowSwitcher does higher-than-centered.

Reply:
Iam fully ok with that (apple vertical offset mimic). Either way looks great. I didn't knew if you placed `ThumbnailPanel` higher on purpose (by using .center() for both panels).

Reply:
In the current commit i had to tweak this behaviour slightly. If the panel to be rendered is huge in its height the vertical offset does not look very good. So I've added a check: If the panel is bigger than 2/3 of the screen, the offset will be not applied.

Is that alright for you?
Thanks!

Reply:
> So I've added a check: If the panel is bigger than 2/3 of the screen, the offset will be not applied.

In my branch I have mimic'ed Apple implementation. They simply position at 75% instead of 50% (0 is bottom, not top). Could you check my implem? I think it's nicer because in your implem there is a big jump in high after you reach a certain size

Reply:
@lwouis I see, your implementation is perfect. Mine was slightly off at different frame sizes. Thanks for that!